backend/app/Actions/Visits/GetPageViewsItemsAction.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Visits;
+
+use App\DataTransformer\Visits\PageViewsItem;
+use App\Entities\Page;
+use App\Repositories\Contracts\TablePageViewsRepository;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+
+final class GetPageViewsItemsAction
+{
+    private $visitRepository;
+
+    public function __construct(TablePageViewsRepository $visitRepository)
+    {
+        $this->visitRepository = $visitRepository;
+    }
+
+    public function execute(GetPageViewsItemsRequest $request): GetPageViewsItemsResponse
+    {
+        $from = $request->startDate();
+        $to = $request->endDate();
+        $pagesId = Page::where('website_id', Auth::user()->website->id)->pluck('id')->toArray();
+
+        $items = [];
+
+        foreach ($pagesId as $id) {
+            $all = $this->visitRepository->getCountPageViewsByPage($from, $to, $id);
+            $bounced = $this->visitRepository->getCountBounceRateByPage($from, $to, $id);
+            $bounceRate = ($all === 0) ? 0 : ($bounced/$all*100);
```

Comment:
may be it would be better fetch collections of $all and $bounced and than iterate them instead of  sql queries in 'foreach' ?